ta/mk/ta_dev_kit.mk

```diff
@@ -65,9 +76,24 @@ clean:
 
 subdirs = .
 include  $(ta-dev-kit-dir)/mk/subdir.mk
+
+#the build target is ta
+ifneq ($(binary),)
 vpath %.c $(ta-dev-kit-dir)/src
 srcs += user_ta_header.c
+endif
 
 include  $(ta-dev-kit-dir)/mk/gcc.mk
 include  $(ta-dev-kit-dir)/mk/compile.mk
+ifneq ($(binary),)
 include  $(ta-dev-kit-dir)/mk/link.mk
+else
+ifneq ($(libname),)
+all: $(libname).a
+cleanfiles += $(libname)
+
+$(libname).a: $(objs)
```

Comment:
There seems to be some confusion here on whether `$(libname)` has the `.a` suffix or not.